packages/preferences-api/src/common/preference-contribution.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { inject, injectable, named } from "inversify";
+import { ContributionProvider, ILogger } from '@theia/core/lib/common';
+import * as ajv from 'ajv';
+
+
+export const PreferenceContribution = Symbol("PreferenceContribution");
+export interface PreferenceContribution {
+    readonly schema: PreferenceSchema;
+}
+
+export const PreferenceSchema = Symbol("PreferenceSchema");
+
+export interface PreferenceSchema {
+    properties: {
+        [name: string]: object
+    }
```

Comment:
you can add index signature here to make code work in other places: `[name: string]: Object`